crates/biome_parser/src/lexer.rs

```diff
@@ -270,107 +326,59 @@ impl LookaheadToken {
     }
 }
 
-impl From<&LexerCheckpoint> for LookaheadToken {
-    fn from(checkpoint: &LexerCheckpoint) -> Self {
+impl<Kind: SyntaxKind> From<&LexerCheckpoint<Kind>> for LookaheadToken<Kind> {
+    fn from(checkpoint: &LexerCheckpoint<Kind>) -> Self {
         LookaheadToken {
             kind: checkpoint.current_kind,
             flags: checkpoint.current_flags,
         }
     }
 }
 
-#[cfg(test)]
-mod tests {
-    use super::BufferedLexer;
-    use crate::lexer::{LexContext, Lexer, TextRange, TextSize};
-    use biome_js_syntax::JsSyntaxKind::{JS_NUMBER_LITERAL, NEWLINE, WHITESPACE};
-    use biome_js_syntax::T;
-
-    #[test]
-    fn without_lookahead() {
-        let lexer = Lexer::from_str("let a\n = 5");
-        let mut buffered = BufferedLexer::new(lexer);
-
-        buffered.next_token(LexContext::default());
-        assert_eq!(buffered.current(), T![let]);
-        assert!(!buffered.has_preceding_line_break());
-        assert_eq!(
-            buffered.current_range(),
-            TextRange::at(TextSize::from(0), TextSize::from(3))
-        );
-
-        assert_eq!(buffered.next_token(LexContext::default()), WHITESPACE);
-        assert_eq!(buffered.next_token(LexContext::default()), T![ident]);
-        assert_eq!(buffered.next_token(LexContext::default()), NEWLINE);
-        assert_eq!(buffered.next_token(LexContext::default()), WHITESPACE);
-        assert_eq!(buffered.next_token(LexContext::default()), T![=]);
-        assert!(buffered.has_preceding_line_break());
-        assert_eq!(buffered.next_token(LexContext::default()), WHITESPACE);
-        assert_eq!(
-            buffered.next_token(LexContext::default()),
-            JS_NUMBER_LITERAL
-        );
-        assert_eq!(buffered.next_token(LexContext::default()), T![EOF]);
+/// Stores the state of the lexer so that it may later be restored to that position.
+#[derive(Debug, Clone)]
+pub struct LexerCheckpoint<Kind: SyntaxKind> {
```

Comment:
I'll try to do it in other places, but I'm not sure how to do it here, because `LexerCheckpoint` has a `current_kind`.
The only way is to replace the struct and with the trait. Do you think it's a proper way to handle this?

Reply:
I see. Let's figure it out in another PR. This PR is already fat enough, let's not put more meat on the fire